filebeat/input/filestream/prospector.go

```diff
@@ -267,11 +267,12 @@ func (p *fileProspector) onRename(log *logp.Logger, ctx input.Context, fe loginp
 	} else {
 		// update file metadata as the path has changed
 		var meta fileMeta
-		err := s.FindCursorMeta(src, meta)
+		err := s.FindCursorMeta(src, &meta)
```

Comment:
This seems like it was an easy mistake to make. How hard is it to add a test that would have caught this?

Reply:
Not very easy with the current implementation because the test would have to use our real [`sourceStore`](https://github.com/elastic/beats/blob/33a9e668de74e715e33ef7f45e7aefd257c8ecf9/filebeat/input/filestream/internal/input-logfile/store.go#L37) because this is the one implementing the [`FindCursorMeta`](https://github.com/elastic/beats/blob/33a9e668de74e715e33ef7f45e7aefd257c8ecf9/filebeat/input/filestream/internal/input-logfile/store.go#L172-L175) that ends up having the issue with non-pointer types.

I could change the mock we use making it test for a pointer on `FindCursorMeta`, it feels reasonable enough because `FindCursorMeta` is expected to mutate its second parameter.